relay_timer_pool.go

```diff
@@ -81,6 +83,11 @@ func (tp *relayTimerPool) Get() *relayTimer {
 
 // Put returns a relayTimer back to the pool.
 func (tp *relayTimerPool) Put(rt *relayTimer) {
+	if tp.verifyPool {
+		// If we are trying to verify correct pool behavior, then we don't release
+		// the timer, and instead ensure no methods are called after being released.
+		return
+	}
```

Comment:
Correct me if I'm wrong: this approach relies on the fact that most of the relay timer's methods assume that they'll only be called once, so not returning timers to the pool and using them concurrently will trigger races. That's not true of the `Stop` method, though - should we also add some safety there?

Reply:
Oh wow, I completely missed a commit when making this branch off the previous change!

There's a `verifyNotReleased` method which is called by all the exported methods and `OnTimer` which will verify `!released`. Adding the commit now, sorry about that!